src/kv/kv.hpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * Canary - A free and open-source MMORPG server emulator
+ * Copyright (©) 2019-2022 OpenTibiaBR <opentibiabr@outlook.com>
+ * Repository: https://github.com/opentibiabr/canary
+ * License: https://github.com/opentibiabr/canary/blob/main/LICENSE
+ * Contributors: https://github.com/opentibiabr/canary/graphs/contributors
+ * Website: https://docs.opentibiabr.com/
+ */
+
+#pragma once
+
+#include <string>
+#include <mutex>
+#include <initializer_list>
+#include <parallel_hashmap/phmap.h>
+#include <optional>
+
+#include "lib/logging/logger.hpp"
+#include "value_wrapper.hpp"
+
+class KVBackend {
+public:
+	virtual std::optional<ValueWrapper> load(const std::string &key) = 0;
```

Comment:
Maybe load and save could be templates? Then we don't automatically bind it to protobuf value wrapper. There might be simpler use cases where we could just want to do a KV of integer or so?

Reply:
ValueWrapper is not protobuf actually, it's a c++ variant. There is indeed a protobuf version of it, but that is not actually a dependency here.

Reply:
Oh I missed that, makes sense.